pkg/tiup/deploy.go

```diff
@@ -63,6 +63,60 @@ func IgnoreClusterNotExist(err error) error {
 	return err
 }
 
+func setPumpConfig(spec *tiupSpec.Specification, pumpConfig string, index int) error {
+	unmarshalPumpConfigToMap := func(data []byte, object *map[string]interface{}) error {
+		err := yaml.Unmarshal(data, object)
+		return err
+	}
+	var (
+		config = make(map[string]interface{}, 0)
+	)
+	for _, item := range []struct {
+		object *map[string]interface{}
+		config string
+	}{{
+		&config,
+		pumpConfig,
+	}} {
+		err := unmarshalPumpConfigToMap([]byte(item.config), item.object)
+		if err != nil {
+			return err
+		}
+	}
+	spec.PumpServers[index].Config = config
+	return nil
+}
+
+func setDrainerConfig(spec *tiupSpec.Specification, drainerConfig string, index int, clusterIPMaps ...map[string]string) error {
+	unmarshalDrainerConfigToMap := func(data []byte, object *map[string]interface{}) error {
+		err := yaml.Unmarshal(data, object)
+		return err
+	}
+	var (
+		config = make(map[string]interface{}, 0)
+	)
+	for _, item := range []struct {
+		object *map[string]interface{}
+		config string
+	}{{
+		&config,
+		drainerConfig,
+	}} {
+		err := unmarshalDrainerConfigToMap([]byte(item.config), item.object)
+		if err != nil {
+			return err
+		}
+	}
+	// transfer
+	if clusterIPMaps != nil {
+		if strings.HasPrefix(config["syncer.to.host"].(string), "naglfar://") {
```

Comment:
parsing and transforming protocol of `syncer.to.host` should put before `if clusterIPMaps != nil`, and you'd better ensure that the naglfar node exists in `clusterIPMaps`.

Reply:
Install, update, delete will use MakeClusterManager(BuildSpecification), I think this scenes and other maybe all need drainers.Some scenes don't need parse "naglfar://n1", but "naglfar://n1" must be parsed correctly when clusterIPMaps is not empty.I will change it as follow.
```go
// transfer
	if clusterIPMaps != nil {
		clusterIP := clusterIPMaps[0][strings.Split(config["syncer.to.host"].(string), "://")[1]]
		if clusterIP == "" {
			return fmt.Errorf("syncer.to.host is not existed")
		} else {
			config["syncer.to.host"] = clusterIP
		}
	}
```